packages/edit-site/src/components/editor/index.js

```diff
@@ -152,12 +152,36 @@ export default function Editor() {
 	// action in <URlQueryController> from double-announcing.
 	useTitle( hasLoadedPost && title );
 
-	if ( ! hasLoadedPost ) {
-		return <CanvasSpinner />;
-	}
+	const { hasResolvingSelectors } = useSelect( ( select ) => {
+		return {
+			hasResolvingSelectors: select( coreStore ).hasResolvingSelectors(),
+		};
+	} );
+	const [ loaded, setLoaded ] = useState( false );
+	const timeoutRef = useRef( null );
+
+	useEffect( () => {
+		if ( ! hasResolvingSelectors && ! loaded ) {
+			clearTimeout( timeoutRef.current );
+			timeoutRef.current = setTimeout( () => {
+				setLoaded( true );
+			}, 1000 );
+		}
+
+		if ( hasResolvingSelectors && timeoutRef.current ) {
+			clearTimeout( timeoutRef.current );
+		}
+
+		return () => {
+			clearTimeout( timeoutRef.current );
+		};
```

Comment:
We can just move the cleanup function inside the if statement where we set the timeout, so that we don't need the second if statement.

Reply:
Agreed, moved in 8f5790ea5e0cca3f8e1d76271c0039ed6d8a57e2